lineid_plot.py

```diff
@@ -418,10 +426,11 @@ def plot_line_ids(wave, flux, line_wave, line_label1, label1_size=None,
     # Redraw the boxes at their new x location.
     for i in range(nlines):
         box = ax.texts[i]
-        if hasattr(box, 'xyann'):
```

Comment:
The if/else statement doesn't do anything and we can remove it. 

The tests in test_lineid_plot.py do not give any errors. Can you post the call that raised exception?

Your code also passes all tests; so I am guessing that the matplotlib versions included in the tests (1.4 and 1.5) both have xyann. Having the test would make the code work happily in previous versions also, even though we are not explicitly testing these versions.

If we can't reproduce the error then I would prefer to have the hasattr check.